### ISSUE TITLE
List of namespaces to scan for injecting sidecars

### ISSUE THREAD

**Issue by objectiser** [jaegertracing/jaeger-operator#929]
Currently the operator is generally installed cluster wide, with WATCH_NAMESPACE left empty to mean scan for CRs in any namespace.

The sidecar injection also uses the WATCH_NAMESPACE to determine which namespaces it should check.

However if we want to only install Jaeger Operator in a particular namespace, this also means the sidecar injection will only check for deployments in that namespace.

The set of namespaces in which a Jaeger instance may be created will in general be different to the set of namespaces in which traced services will be deployed. Therefore, when the operator is only deployed in a particular namespace (with defined WATCH_NAMESPACE), we need an optional way to configure the operator (for a particular CR) to scan other namespaces (subject to relevant permissions).

One suggestion would be to include a namespace list in the CR itself.

**Comment by pavolloffay** [comment]
If am not mistaken the namespaces in the CR will have to be also specified in the `WATCH_NAMESPACE`, however it can be just a subset of those namespaces? Am I right?

**Comment by jpkrohling** [comment]
The list might not be related at all: we might want to `WATCH_NAMESPACE=observability` for Jaeger CRs, but inject sidecars in `my-project`.

**Comment by pavolloffay** [comment]
I think I got confused with your docs PR. 

So basically `WATCH_NAMESPACE` watches the namespaces but to actually make it work one has to create `RoleBindings` (RB) for each namespace (if not using cluster-wide permissions).

Then the list of inject namespaces in the RC will also require RB for each namespace. I think I got it now.

**Comment by pavolloffay** [comment]
Btw. maistra (isito operator) has a separate CR which defines watched namespaces

**Comment by jpkrohling** [comment]
Right, the operator will need permissions in each namespace listed in the new field in the CR to change deployments (get,list,watch,update, but probably not create,delete), so that it gets to inject sidecars.

To watch for Jaeger CRs, it needs a different set of permissions that apply only to the list of namespaces from `WATCH_NAMESPACE`.

**Comment by jpkrohling** [comment]
Right, the operator will need permissions in each namespace listed in the new field in the CR to change deployments (get,list,watch,update, but probably not create,delete), so that it gets to inject sidecars.

To watch for Jaeger CRs, it needs a different set of permissions that apply only to the list of namespaces from `WATCH_NAMESPACE`.

**Comment by esnible** [comment]
I create https://github.com/jaegertracing/jaeger-operator/pull/1551 because it bothered me that all the namespaces were being watched for injection.  However, my implementation used WATCH_NAMESPACE.  After reading this, perhaps we need a separate INJECT_NAMESPACE environment configuration instead?
